src/main/java/org/sciborgs1155/robot/pivot/Pivot.java

```diff
@@ -0,0 +1,200 @@
+                log -> {
+                  log.motor("pivot")
+                      .angularPosition(Radians.of(pivot.getPosition().getRadians()))
+                      .angularVelocity(RadiansPerSecond.of(pivot.getVelocity()));
+                },
+                this,
+                "pivot"));
+
+    pivotPID.setTolerance(POSITION_TOLERANCE.in(Radians));
+    climbPID.setTolerance(POSITION_TOLERANCE.in(Radians));
+
+    SmartDashboard.putData("pivot quasistatic forward", quasistaticForward());
+    SmartDashboard.putData("pivot quasistatic backward", quasistaticBack());
+    SmartDashboard.putData("pivot dynamic forward", dynamicForward());
+    SmartDashboard.putData("pivot dynamic backward", dynamicBack());
+
+    setDefaultCommand(runPivot(() -> STARTING_ANGLE).repeatedly());
+  }
+
+  /**
+   * Smoothly angle the pivot to a desired goal using a {@link ProfiledPIDController}.
+   *
+   * @param goalAngle The position to move the pivot to.
+   * @return The command to set the pivot's angle.
+   */
+  public Command runPivot(Supplier<Rotation2d> goalAngle) {
+    return run(() -> {
+          double feedback =
+              pivotPID.calculate(pivot.getPosition().getRadians(), goalAngle.get().getRadians());
+          double feedforward =
+              pivotFeedforward.calculate( // add pi to measurement to account for alternate angle
+                  pivotPID.getSetpoint().position + Math.PI, pivotPID.getSetpoint().velocity);
+          pivot.setVoltage(feedback + feedforward);
+        })
+        .until(pivotPID::atGoal)
+        .withName("running pivot");
+  }
+
+  public Command manualPivot(InputStream stickInput) {
+    return runPivot(
+        () -> {
+          double velocity = stickInput.get() * PivotConstants.MAX_VELOCITY.in(RadiansPerSecond);
+          double periodMovement = Constants.PERIOD.in(Seconds) * velocity;
+          double draftSetpoint = periodMovement + pivot.getPosition().getRadians();
+          double setpoint =
+              Math.max(Math.min(MAX_ANGLE.getRadians(), draftSetpoint), MIN_ANGLE.getRadians());
+          return Rotation2d.fromRadians(setpoint);
+        });
+  }
+
+  public Command climb(Supplier<Rotation2d> goalAngle) {
+    return run(() -> {
+          double feedback =
+              climbPID.calculate(pivot.getPosition().getRadians(), goalAngle.get().getRadians());
+          double feedforward =
+              climbFeedforward.calculate( // add pi to measurement to account for alternate angle
+                  climbPID.getSetpoint().position + Math.PI, climbPID.getSetpoint().velocity);
+          pivot.setVoltage(feedback + feedforward);
+        })
+        .until(pivotPID::atGoal)
+        .withName("climbing");
+  }
+
+  @Log.NT
+  public State getGoal() {
+    return pivotPID.getGoal();
+  }
+
+  @Log.NT
+  public Rotation2d getPosition() {
+    return pivot.getPosition();
+  }
+
+  @Log.NT
+  public boolean atGoal() {
+    return pivotPID.atGoal();
+  }
+
+  @Log.NT
```

Comment:
remove the `State` `getSetpoint` and rename `getSetpointRadians` to `getSetpoint